### PR TITLE
Fix eventService memory leaks

### PR DESCRIPTION
**Pull request by nzdev**
### Prerequisites

- [x] I have added steps to test this contribution in the description below

If there's an existing issue for this PR then this fixes N/A

### Description
What?
Fix memory leaks by unbinding event from $rootscope.

Why?
Memory leaks in the browser when events are not unbound from the $rootscope.

How to test?
Ensure memory freed when navigating the backoffice ui.

### CODE REVIEW

**Comment by nzdev** [comment]
tinymce.service.js also appears to be leaking via events, leaving tinymce content in memory

File: src/Umbraco.Web.UI.Client/src/common/services/editor.service.js [hunk 1]
```diff
@@ -173,13 +173,14 @@ When building a custom infinite editor view you can use the same components as a
...
         // events for backdrop
-        eventsService.on("appState.backdrop", function (name, args) {
+        var backgroundEvent = eventsService.on("appState.backdrop", function (name, args) {
             if (args.show === true) {
                 blur();
             } else {
                 focus();
             }
         });
+
```
Comment (nzdev) [inline src/Umbraco.Web.UI.Client/src/common/services/editor.service.js:183]: no access to $scope, need another approach for destruction

File: src/Umbraco.Web.UI.Client/src/common/services/localization.service.js [hunk 2]
```diff
@@ -326,10 +326,11 @@ angular.module('umbraco.services')
     };
 
     //This happens after login / auth and assets loading
-    eventsService.on("app.authenticated", function () {
+    var appAuthenticated = eventsService.on("app.authenticated", function () {
         resourceFileLoadStatus = "none";
         resourceLoadingPromise = [];
     });
+
```
Comment (nzdev) [inline src/Umbraco.Web.UI.Client/src/common/services/localization.service.js:333]: no access to $scope, need another approach for destruction

File: src/Umbraco.Web.UI.Client/src/common/directives/components/content/umbvariantcontenteditors.directive.js [hunk 3]
```diff
@@ -201,6 +203,7 @@
 
             $location.search({"cculture": culture, "csegment": vm.editors[0].content.segment});
             splitViewChanged();
+            //unbindSplitViewRequest();
```
Comment (nzdev) [inline src/Umbraco.Web.UI.Client/src/common/directives/components/content/umbvariantcontenteditors.directive.js:206]: remove commented out line

**Comment by nzdev** [comment]
https://www.toptal.com/angular-js/top-18-most-common-angularjs-developer-mistakes

File: src/Umbraco.Web.UI.Client/src/common/services/editor.service.js (hunk 1 above)
Comment (nathanwoulfe) [reply src/Umbraco.Web.UI.Client/src/common/services/editor.service.js:183]: @nzdev service is a singleton, so no need to worry about tear down

**Review by nathanwoulfe — APPROVED** [review]
Thanks @nzdev, I'd noticed memory spikes a while back when opening multiple infinite editors, this should help alleviate that issue. I've reverted the changes in the services as it's not required due to the lifespan of those object.

**Pull request merged**